src/generic/armcm_timer.c

```diff
@@ -95,6 +95,9 @@ timer_init(void)
 {
     // Enable Debug Watchpoint and Trace (DWT) for its 32bit timer
     CoreDebug->DEMCR |= CoreDebug_DEMCR_TRCENA_Msk;
+#if CONFIG_MACH_STM32F7
+    DWT->LAR = 0xC5ACCE55;
+#endif
```

Comment:
Can this be moved into stm32f7.c instead?  That is, add something like the following to `armcm_main()`:
```c
   CoreDebug->DEMCR |= CoreDebug_DEMCR_TRCENA_Msk;
   DWT->LAR = 0xC5ACCE55;
```

I'm a little leery of adding architecture specific code to the generic code.

Reply:
I would argue this is better as-is. The DWT register unlock is needed for DWT->CTRL access one line below. Setting that register out of context would make this harder than needed to track the problem.

Marlin does that too: https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.1.x/Marlin/src/HAL/shared/Delay.cpp#L90-L92
Betaflight (drone flight controller) does that as well: https://github.com/betaflight/betaflight/blob/master/src/main/drivers/system.c#L79-L89

Reply:
The convention is to not put chip specific code in the `generic/` directory (`grep CONFIG_MACH_ src/generic/*`).  I'd prefer to keep that convention as I fear the alternative would quickly devolve into a mess of ifdefs in the "generic" code.

Possible alternatives are:
1. Place the two line override in stm32f7.c (as proposed earlier).
2. Copy generic/armcm_timer.c to a new stm32/stm32f7_timer.c .  That's a lot of dup'ed code though and seems not worth it.
3. Use an architecture specific ifdef in armcm_timer.c (eg, `if (__CORTEX_M >= 7)`).  Unfortunately, this risks introducing a regression on other chips, and if a regression is found a revert would require reverting all of stm32f7.c .  So, seems like putting the code in stm32f7.c is less risky (even if only for the short-term).

-Kevin

Reply:
Allright, moved it to stm32f7.c